CONTRIBUTING_GO.md

```diff
@@ -161,6 +164,47 @@ reasons why this is _the right thing to do_:
 License: MIT
 Signed-off-by: Juan Benet <juan@ipfs.io>
 ```
+
+## Contributor Principles
+
+These principles are a living document and should be updated as-needed to better serve the go-ipfs community.
+
+In order of precedence:
+
+1. **Build the distributed peer-to-peer web.** This is our objective and aim - everything we do is in service of this goal and we only compromise our other principles if needed to achieve this.
+2. **Don't break the IPFS ecosystem.** This isn't "don't break protocols"; we do occasionally break protocols and APIs (with care) when we know that users won't notice it. However, users trust us to not break their workflows/projects. If we do, we'll lose that trust and those users.
+3. **Take ownership.** If you're working on something, own it and see it through. Given that (a) we don't have managers and (b) we don't work synchronously, coordination can be difficult. To make up for this, if you want something to happen, you have to make it happen (that may mean calling a meeting, bugging people, fixing unrelated issues, designing alternative solutions, etc.).
+4. **Ask for forgiveness, not permission.** To enable P.3, if you're having trouble getting people to move on something, err on the side of getting things done (bounded by P.1 and P.2). Note: This doesn't apply when you *know* you'll need to ask for forgiveness... use your best judgment.
+
+## Code Change Policy
+
+The go code change policy must balance "not breaking things" with "getting things done" and should be read in that spirit. Use your best judgment when making changes and PRs; these rules are not set in stone.
+
+**How can the codebase be changed?**
+
+Except in special cases (and mass gx-workspace updates), all changes go through PRs.
+
+* Larger changes should be discussed in an issue *before* work is started to avoid wasted effort, tears, and dead fairies.
+* WIP (work in progress), RFC (request for comments) changes should be marked as such. Not doing so can rouse the review gremlins.
```

Comment:
```suggestion
* WIP (work in progress), RFC (request for comments) changes should be marked as such. Not doing so is inconsiderate to reviewers, who may think your PR is ready for review. Consider that they may spend time reviewing it, before it is actually ready, so it's best to inform everyone that something is not ready.
```
Okay this isn't concise, but I think it conveys the intention. Maybe it's helpful.

Reply:
I've simplified your change a bit and made it clear that it's not exactly inconsiderate, just frustrating (PR authors are *usually* trying to help).